packages/circuit-ui/components/SidePanel/SidePanel.tsx

```diff
@@ -15,147 +15,171 @@
-      labelledby: headerAriaId,
-    },
-    onRequestClose: onBack || onClose,
-    bodyOpenClassName: '',
-    /**
-     * react-modal relies on document.activeElement to return focus after the modal is closed.
-     * Safari and Firefox don't set it properly on button click (see https://github.com/reactjs/react-modal/issues/858 and https://github.com/reactjs/react-modal/issues/389).
-     * Returning the focus to document.body or to the focus-root can cause unwanted page scroll.
-     * Preventing scroll on focus would provide better UX for mouse users and shouldn't cause any side effects for assistive technology users.
-     */
-    preventScroll: true,
-  };
+      useEffect(() => {
+        // trigger animation on opening
+        triggerAnimationRef.current(true);
+      }, [triggerAnimationRef]);
 
-  const content = (
-    <div className={classes.base} onScroll={handleScroll}>
-      <Header
-        backButtonLabel={backButtonLabel}
-        closeButtonLabel={closeButtonLabel}
-        headline={headline}
-        id={headerAriaId}
-        onBack={onBack}
-        onClose={onClose}
-        isSticky={isHeaderSticky}
-      />
-      <div className={classes.content}>
-        {isFunction(children) ? children({ onBack, onClose }) : children}
-      </div>
-    </div>
-  );
+      useEffect(
+        () => () => {
+          onCloseEnd?.();
+        },
+        [onCloseEnd],
+      );
 
-  if (isMobile) {
-    return (
-      <MobileSidePanel
-        {...defaultProps}
-        {...props}
-        isBottomPanelClosing={isBottomPanelClosing}
-        isStacked={isStacked}
-      >
-        {content}
-      </MobileSidePanel>
-    );
-  }
+      const content = (
+        <div onScroll={handleScroll}>
+          <Header
+            backButtonLabel={backButtonLabel}
+            closeButtonLabel={closeButtonLabel}
+            headline={headline}
+            id={headerAriaId}
+            onBack={onBack ? () => triggerAnimation(false, onBack) : undefined}
+            onClose={() => triggerAnimation(false, onClose)}
+            isSticky={isHeaderSticky}
+          />
+          <div className={classes.content}>
+            {isFunction(children) ? children({ onBack, onClose }) : children}
+          </div>
+        </div>
+      );
 
-  return (
-    <DesktopSidePanel
-      {...defaultProps}
-      {...props}
-      isInstantOpen={isInstantOpen}
-    >
-      {content}
-    </DesktopSidePanel>
-  );
-};
+      return (
+        <Dialog
+          {...rest}
+          ref={ref}
+          open={open}
+          isModal={isMobile}
+          aria-labelledby={headerAriaId}
+          animationDuration={animationDuration}
+          className={clsx(classes.base, animationClass, className)}
+          onCloseStart={() => triggerAnimation(false, onBack || onClose)}
+          preventOutsideClickClose={true}
+          preventEscapeKeyClose={preventEscapeKeyClose}
+          hideCloseButton
+        >
+          {content}
+        </Dialog>
```

Comment:
So nice to see the mobile and desktop views unified 🤩  

You could move the `content` JSX inside the Dialog since there's no more reason to separate them.